eth2/simulation/src/simulation.rs

```diff
@@ -15,37 +17,205 @@ where
 }
 
 impl<T: EthSpec> Simulation<T> {
+    pub fn new() -> Self {
+        Self {
+            store: Store::new(),
+        }
+    }
+
     /// Add a new execution environment, return EE index
-    fn create_execution_environment(a: args::CreateExecutionEnvironment) -> EEIndex {
-        unimplemented!();
+    pub fn create_execution_environment(
+        &mut self,
+        a: args::CreateExecutionEnvironment,
+    ) -> Result<EEIndex> {
+        // Create EE struct from args
+        let vl_wasm_code =
+            VariableList::new(a.wasm_code).map_err(|_| Error::MaxLengthExceeded {
+                what: format!("wasm_code"),
+            })?;
+        let ee = ExecutionEnvironment {
+            initial_state: a.initial_state,
+            wasm_code: vl_wasm_code,
+        };
+
+        // Add EE code to beacon chain
+        self.store
+            .current_beacon_state
+            .execution_environments
+            .push(ee)
+            .map_err(|_| Error::MaxLengthExceeded {
+                what: format!("number of execution environments"),
+            })?;
+
+        // For each shard, add the initial state to the shard
+        for shard_state in self.store.current_beacon_state.shard_states.iter_mut() {
```

Comment:
I think I'd like to see a little more encapsulation here. Instead of reaching through multiple levels of struct's internal fields, create some helper methods. Should reduce the amount of duplicate code (especially error conversion.)

Reply:
Thanks for the comment! I don't think I agree for this particular case, though it's possible I'm misunderstanding your point here.  What kind of helper function do you envision?

My thoughts:
* I don't expect any code duplication from this sort of thing -- chaining properties like this seems normal and readable, and `current_beacon_state` will always have `shard_states`, etc.
* Not sure what you mean by error conversion -- there's no error conversion here. Let me know what I'm missing.

I could see this being applicable in other contexts, is there another line or example that you think could benefit from helper methods?